zypp-plugin/snapper_zypp_plugin.cc

```diff
@@ -0,0 +1,349 @@
+// getenv
+#include <stdlib.h>
+// getppid
+#include <sys/types.h>
+#include <unistd.h>
+// fnmatch
+#include <fnmatch.h>
+
+#include <boost/regex.hpp>
+#include <boost/format.hpp>
+using boost::format;
+
+#include <iostream>
+#include <map>
+#include <string>
+using namespace std;
+
+#include <rapidjson/document.h>
+
+#include "dbus/DBusConnection.h"
+#include "snapper/Exception.h"
+using snapper::Exception;
+using snapper::CodeLocation;
+#include "client/commands.h"
+#include "client/errors.h"
+#include "snapper/Log.h"
+#include "snapper/XmlFile.h"
+
+#include "zypp_commit_plugin.h"
+
+ostream& operator <<(ostream& os, set<string> ss) {
+    bool seen_first = false;
+    os << '{';
+    for(auto s: ss) {
+	if (seen_first) {
+	    os << ", ";
+	}
+	os << s;
+	seen_first = true;
+    }
+    os << '}';
+    return os;
+}
+
+class ProgramOptions {
+public:
+    string plugin_config;
+    string snapper_config;
+    DBusBusType bus;
+
+    ProgramOptions()
+    : plugin_config("/etc/snapper/zypp-plugin.conf")
+    , snapper_config("root")
+    , bus(DBUS_BUS_SYSTEM)
+    {
+	const char * s;
+
+	s = getenv("SNAPPER_ZYPP_PLUGIN_CONFIG");
```

Comment:
The env. var. should be explained, esp. that they are only used for debugging AFAIS.

Reply:
Does the new comment of class ProgramOptions look good?

Reply:
I assume it is good enough.